index.js

```diff
@@ -86,22 +72,25 @@ const buildProject = async () => {
   }
   if (isMindmapUpdated) {
     console.log("Building Mindmap");
-    await executeCmd(`cd ${process.env.PROJECT_PATH}/mindmap && npm run build`);
+    await initiateBuild("mindmap",'npm run build');
     mindmapBuildTime = currentTime;
     isMindmapUpdated = false;
-
-    await executeCmd(`cp -r ${process.env.PROJECT_PATH}/mindmap/dist/ ${process.env.DIST_PATH}`);
   }
 
   if (isDocumentationWebsiteUpdated) {
     console.log("Building Documentation Website");
-    await executeCmd(`cd  ${process.env.PROJECT_PATH}/documentation-website && npm run build`);
+    await initiateBuild("documentation-website","npm run build");
     documentationWebsiteBuildTime = currentTime;
     contributorsBuildTime = currentTime;
     isDocumentationWebsiteUpdated = false;
-
-    await executeCmd(`cp -r ${process.env.PROJECT_PATH}/documentation-website/dist/ ${process.env.DIST_PATH}`);
   }
 
   return 200, "Build has been created.";
 };
+
+const initiateBuild = async (projectName,command) =>{
+  await executeCmd(`cd ${process.env.ROOT_PATH}/${projectName} && git pull`);
+  await executeCmd(`cd ${process.env.ROOT_PATH}/${projectName} && npm ci`);
+  await executeCmd(`cd ${process.env.ROOT_PATH}/${projectName} && ${command}`);
+  await executeCmd(`cp -r ${process.env.DEST_PATH}/${projectName}/dist/ ${process.env.DEST_PATH}/${projectName}/`);
+}
```

Comment:
Good idea and implementation, it just doesn't fix the issue with multiple destination paths and root paths. Those have to be project name specific.